web-client/src/providers/socket.js

```diff
@@ -22,34 +22,34 @@ export const socketProvider = ({ socketRouter }) => {
 
   const start = () => {
     const token = app.getState('token');
+    if (!socket) {
+      return new Promise((resolve, reject) => {
+        try {
+          socket = createWebSocketClient(token);
+          socket.onmessage = socketRouter(app);
+          socket.onerror = error => {
+            applicationContext.logger.error('Websocket error detected', error);
+            return reject(error);
+          };
 
-    if (socket && socket.close) {
-      stopSocket();
-    }
-
-    return new Promise((resolve, reject) => {
-      try {
-        socket = createWebSocketClient(token);
-        socket.onmessage = socketRouter(app);
-        socket.onerror = reject;
-
-        socket.onopen = () => {
-          // the socket needs to be open for a short period or it could miss the first message
-          setTimeout(() => {
-            resolve();
-          }, 300);
-        };
-      } catch (e) {
-        if (applicationContext) {
-          applicationContext.logger.error(
-            'Failed to establish WebSocket connection',
-            e,
-          );
+          socket.onopen = () => {
+            // the socket needs to be open for a short period or it could miss the first message
+            setTimeout(() => {
+              resolve();
+            }, 300);
+          };
+        } catch (e) {
+          if (applicationContext) {
+            applicationContext.logger.error(
+              'Failed to establish WebSocket connection',
+              e,
```

Comment:
Just because we have seen [some other strange behavior passing the Error object along](https://github.com/ustaxcourt/ef-cms/pull/1171/files#diff-f5f2fff53bc2682640ec3a00b96baf967e5c45c6e2c428f8ec06936f0f3edd63R276), can you encapsulate `e` in an object

```
applicationContext.logger.error(
  'Failed to establish WebSocket connection',
  { e },
);
```